tests/nix-profile.sh

```diff
@@ -120,3 +120,19 @@ nix profile install "$flake1Dir^man"
 (! [ -e $TEST_HOME/.nix-profile/bin/hello ])
 [ -e $TEST_HOME/.nix-profile/share/man ]
 (! [ -e $TEST_HOME/.nix-profile/include ])
+
+# test priority
+nix profile remove 0
+
+# Make another flake.
+flake2Dir=$TEST_ROOT/flake2
+printf World > $flake1Dir/who
+cp -r $flake1Dir $flake2Dir
+printf World2 > $flake2Dir/who
+
+nix profile install $flake1Dir
+[[ $($TEST_HOME/.nix-profile/bin/hello) = "Hello World" ]]
+nix profile install $flake2Dir --priority 100
+[[ $($TEST_HOME/.nix-profile/bin/hello) = "Hello World" ]]
+nix profile install $flake2Dir --priority 0
+[[ $($TEST_HOME/.nix-profile/bin/hello) = "Hello World2" ]]
```

Comment:
It doesn't, I'll elaborate below

Reply:
Right now installing just adds an entry to the manifest.json of the profile and rebuilds it:
It looks like this after re-installing with priority 100.

manifest.json:
[{flake1, priority: 5},{flake2, priority: 100},{flake2, priority: 0},{flake2, priority: 100}]

And the priority: 0 of flake2 get's respected

This is weird.

I'd add deduplication logic into `nix profile install` or src/libstore/builtins/buildenv.cc:118 buildProfile . Not sure if anyone relies on knowing the history though.

Reply:
Mh thanks for investigating.

That all works a bit magically because
1. There’s [a check](https://github.com/NixOS/nix/blob/bf89cd95a4af35ab15f7fad3186c8f6190f87c84/src/libstore/builtins/buildenv.cc#L124) ensuring that each store path is considered at most once
2. The profile elements are [sorted by increasing priority](https://github.com/NixOS/nix/blob/bf89cd95a4af35ab15f7fad3186c8f6190f87c84/src/libstore/builtins/buildenv.cc#L141-L143), so the one with the lowest priority is the one used

So yes, there probably should be some deduplication logic in `nix profile` (that would properly handle changing the priority of an existing package, and reflect that into the history, and everything). But that doesn’t have to happen right in this PR (it’s a bit orthogonal, and getting the details right might be a bit tricky)

Reply:
(I wouldn’t mind you opening a separate issue about this though :) )